src/controllers/admin/donate/donate.js

```diff
@@ -0,0 +1,51 @@
+const { join } = require('path');
+const donates = require('../../../database/queries/admin/getQuery');
+const insertOne = require('../../../database/queries/admin/insertOne');
+
+exports.getDonates = (request, response) => {
+  donates('donates', {})
+    .then((card) => {
+      response.status(200).render('admin/donates', {
+        js: ['domUyils', join('admin', 'donateDom')],
+        css: [join('admin', 'donate')],
+        layout: 'admin',
+        card,
+      });
+    }).catch(() => {
+      response.status(500).send({
+        msg: 'server error',
+      });
+    });
+};
+
+exports.postDonates = (request, response) => {
+  const {
+    id,
+    obj,
+  } = request.body;
+  if (obj.current) {
+    insertOne('donates', id, obj)
+      .then(() => {
+        response.status(200).send({
+          msg: 'convert from cerrent to done',
+        });
+      })
+      .catch(() => {
+        response.status(500).send({
+          msg: 'server error',
+        });
+      });
+  } else if (obj.done) {
+    insertOne('donates', id, obj)
```

Comment:
@AbdallahAmmar96 Here's another place where you could reduce some duplication, both the if-statement on line 26 and the if-statement here on 38 are doing pretty similar work, just with a different response message, so you could combine them a little bit.

Reply:
thank for this informations